### PR TITLE
dropbox apphook sources

### PR DESCRIPTION
**Pull request by danhsiung**
basic dropbox sources

### CODE REVIEW

File: components/dropbox/dropbox.app.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          ({ entries, has_more, cursor } = files)
+
+          for(entry of entries) {
+            if (entry[".tag"] == "folder") {
+              options.push(entry.path_display)
+            }
+          }
+          // TODO break after a certain number of folders has been found??
+
+          if (has_more = files.has_more) {
```
Comment (tjk) [inline components/dropbox/dropbox.app.js:49]: What is the point of this if? has_more already destructured above

File: components/dropbox/dropbox.app.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            files = await sdk.filesListFolderContinue({ cursor })
+          }
+        } while(has_more)
+        options = options.sort((a, b) => { return a.toLowerCase().localeCompare(b.toLowerCase()) })
+        if (path && path != "") {
```
Comment (tjk) [inline components/dropbox/dropbox.app.js:54]: there is no point of the second part of this conditional (if path = "" it is not truthy)

File: components/dropbox/dropbox.app.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          let parent = path.slice(0, path.lastIndexOf('/'))
```
Comment (tjk) [inline components/dropbox/dropbox.app.js:55]: maybe use https://nodejs.org/api/path.html#path_path_dirname_path ?

Comment (tjk) [reply components/dropbox/dropbox.app.js:55]: idk if trailing slash is handled correctly here?

Comment (tjk) [reply components/dropbox/dropbox.app.js:55]: maybe dropbox never returns that

File: components/dropbox/dropbox.app.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          if (parent == "") parent = "/"
+          options.unshift(parent)
+        }
+        options.unshift(path)
+      } catch (err) {
+        options.push("/")
```
Comment (tjk) [inline components/dropbox/dropbox.app.js:61]: Does this error handling make sense??

File: components/dropbox/dropbox.app.js (hunk 1 above)
Comment (danhsiung) [reply components/dropbox/dropbox.app.js:49]: for the while below

Comment (danhsiung) [reply components/dropbox/dropbox.app.js:49]: good catch

File: components/dropbox/dropbox.app.js (hunk 2 above)
Comment (danhsiung) [reply components/dropbox/dropbox.app.js:54]: removing

File: components/dropbox/dropbox.app.js (hunk 3 above)
Comment (danhsiung) [reply components/dropbox/dropbox.app.js]: done

File: components/dropbox/dropbox.app.js (hunk 4 above)
Comment (danhsiung) [reply components/dropbox/dropbox.app.js:59]: was thinking makes sense to default to the user's home dir

**Comment by danhsiung** [comment]
@tjk did some refactoring today. Any thoughts on current state? Also, did you take a look at the firehose component code (god mode as pd@pipedream.com - https://pipedream.com/sources/dc_zqu3kx)?

**Pull request merged**